core/src/main/java/jenkins/model/ParameterizedJobMixIn.java

```diff
@@ -178,7 +178,7 @@ public final void doBuild(StaplerRequest req, StaplerResponse rsp, @QueryParamet
         }
 
         if (!asJob().isBuildable()) {
-            throw HttpResponses.error(SC_INTERNAL_SERVER_ERROR, new IOException(asJob().getFullName() + " is not buildable"));
+            throw HttpResponses.error(SC_CONFLICT, new IOException(asJob().getFullName() + " is not buildable"));
```

Comment:
> **400 Bad Request** The request could not be understood by the server due to malformed syntax. The client SHOULD NOT repeat the request without modifications.

> **409 Conflict** The request could not be completed due to a conflict with the current state of the resource. This code is only allowed in situations where it is expected that the user might be able to resolve the conflict and resubmit the request. The response body SHOULD include enough

Taken from here: https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

409 seems more close to the situation. What do you guys say? If SC_CONFLICT seems better then, build-token-root should also use 409.